pkg/crd/flatten.go

```diff
@@ -147,6 +147,8 @@ func flattenAllOfInto(dst *apiext.JSONSchemaProps, src apiext.JSONSchemaProps, e
 			dstField.Set(srcField)
 		case "XMapType":
 			dstField.Set(srcField)
+		case "XValidations":
+			dstField.Set(reflect.AppendSlice(dstField, srcField))
```

Comment:
I would flip the order here. I think it's better if the validation from the field is before the type.

Basically this allows to add validations add the start of the validation array when importing external types (~ that would prioritize "local" validations over imported ones)

Reply:
Order flipped in https://github.com/kubernetes-sigs/controller-tools/pull/998/commits/43d3d3a2a4df5c76f8255006d8873f7cc63d9953.